example/lib/ios_popover/_ios_popover_menu.dart

```diff
@@ -0,0 +1,370 @@
+      path
+        ..moveTo(contentRect.centerLeft.dx, arrowCenter - halfOfBase)
+        ..relativeLineTo(-arrowLength, halfOfBase)
+        ..relativeLineTo(arrowLength, halfOfBase);
+    } else if (arrowDirection == ArrowDirection.right) {
+      path
+        ..moveTo(contentRect.centerRight.dx, arrowCenter - halfOfBase)
+        ..relativeLineTo(arrowLength, halfOfBase)
+        ..relativeLineTo(-arrowLength, halfOfBase);
+    } else if (arrowDirection == ArrowDirection.up) {
+      path
+        ..moveTo(arrowCenter - halfOfBase, contentRect.topCenter.dy)
+        ..relativeLineTo(halfOfBase, -arrowLength)
+        ..relativeLineTo(halfOfBase, arrowLength);
+    } else {
+      path
+        ..moveTo(arrowCenter - halfOfBase, contentRect.bottomCenter.dy)
+        ..relativeLineTo(halfOfBase, arrowLength)
+        ..relativeLineTo(halfOfBase, -arrowLength);
+    }
+
+    path.close();
+
+    return path;
+  }
+
+  /// Computes the direction where the arrow should point to.
+  ///
+  /// If [globalFocalPoint] is inside the [menuRect] horizontal bounds, the arrow points up or right. Otherwise,
+  /// the arrow points left or right.
+  ///
+  /// If [allowHorizontalArrow] is `false`, the arrow only points up or down.
+  ArrowDirection _computeArrowDirection(Rect menuRect, Offset globalFocalPoint) {
+    final isFocalPointInsideHorizontalBounds =
+        globalFocalPoint.dx >= menuRect.left && globalFocalPoint.dx <= menuRect.right;
+    if (isFocalPointInsideHorizontalBounds || !allowHorizontalArrow) {
+      if (globalFocalPoint.dy < menuRect.top) {
+        return ArrowDirection.up;
+      }
+      return ArrowDirection.down;
+    } else {
+      if (globalFocalPoint.dx < menuRect.left) {
+        return ArrowDirection.left;
+      }
+      return ArrowDirection.right;
+    }
+  }
+
+  /// Computes the center point of the arrow.
+  ///
+  /// This point can be on the x or y axis, depending on the [direction].
+  double _computeArrowCenter(ArrowDirection direction, Offset focalPoint) {
+    final desiredFocalPoint = _isArrowVertical(direction) //
+        ? focalPoint.dx
+        : focalPoint.dy;
+
+    return _constrainFocalPoint(desiredFocalPoint, direction);
+  }
+
+  /// Computes the (x, y) offset used to paint the menu content inside the popover.
+  ///
+  /// When [direction] is up or left, the content needs to be shifted
+  /// to leave space for the arrow.
+  Offset _computeContentOffset(ArrowDirection direction, double arrowLength) {
+    return Offset(
+      (padding?.left ?? 0) + arrowLength,
+      (padding?.top ?? 0) + arrowLength,
+    );
+  }
+
+  /// Indicates whether or not the arrow points to a vertical direction.
+  bool _isArrowVertical(ArrowDirection arrowDirection) =>
+      arrowDirection == ArrowDirection.up || arrowDirection == ArrowDirection.down;
+
+  /// Minimum focal point for the given [arrowDirection] in which the arrow can be displayed inside the popover bounds.
```

Comment:
As mentioned before, there's no indication of what "minimum" means. Or "maximum". These comments should begin with a framing that makes it clear what we're minimizing or maximizing and how to interpret the return value.

Reply:
It would be the `x` or `y` offset of the arrow, depending on the direction. If the arrow points left or right it would be `y`, otherwise `x`.

What would be a better description?